UI/Panels/Settings/MobiFlightPanel.cs

```diff
@@ -1074,15 +1070,33 @@ private void openToolStripButton_Click(object sender, EventArgs e)
 
         private void OpenFile(TreeNode moduleNode, string fileName)
         {
-            TextReader textReader = new StreamReader(fileName);
-            XmlSerializer serializer = new XmlSerializer(typeof(MobiFlight.Config.Config));
-            MobiFlight.Config.Config newConfig;
-            newConfig = (MobiFlight.Config.Config)serializer.Deserialize(textReader);
-            textReader.Close();
+            var newConfig = MobiFlight.Config.Config.LoadFromFile(fileName);
+            var module = (moduleNode.Tag as MobiFlightModule);
+
+            if (newConfig?.ModuleType != module.Type)
+            {
+                // Old configs don't contain a specified type 
+                // In this case newConfig.ModuleType == "" 
+                var message = string.Format(i18n._tr("uiMessageOpenConfigUnspecificTypeText"), module.Type);
+
+                // New configs have a specific type other than ""
+                if ((newConfig?.ModuleType ?? "") != "")
```

Comment:
It's a bit of a style thing, but I prefer using `String.IsNullOrEmpty()` when testing strings for empty values instead of "".

Reply:
good point. they introduced that

Reply:
It reads better, yet I don't understand why some of these are not simply an extension method.